CRM/Core/Payment/OmnipayMultiProcessor.php

```diff
@@ -657,7 +657,12 @@ public function processPaymentNotification($params) {
     $originalRequest = $_REQUEST;
     $_REQUEST = $params;
     try {
-      $response = $this->gateway->completePurchase($params)->send();
+      if ($this->gateway->supportsAcceptNotification()) {
```

Comment:
Hi @mattwire I've just added this commit to the main repo - I'm afraid I'm cherry picking my way through your commits as time permits rather than assessing as a whole, as there are a lot of issues in there to consider